probe/Dockerfile

```
@@ -1,13 +1,49 @@
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5
+FROM registry.access.redhat.com/ubi8/s2i-base:1-388 AS build
 
-COPY bin/probe /stackrox/
+ARG GO_VERSION=1.18.8
+RUN curl -L --retry 10 --silent --show-error --fail -o /tmp/go.linux-amd64.tar.gz \
+	"https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz" && \
+	tar -C /usr/local -xzf /tmp/go.linux-amd64.tar.gz && \
+	rm -f /tmp/go.linux-amd64.tar.gz
+ENV PATH="/usr/local/go/bin:${PATH}"
 
-EXPOSE 7070
+ARG GOPATH=/go
+ENV GOPATH=${GOPATH}
 
-ENTRYPOINT ["/stackrox/probe"]
+ARG GOCACHE=/go/.cache
+ENV GOCACHE=${GOCACHE}
+
+ARG GOROOT=/usr/local/go
+ENV GOROOT=${GOROOT}
+
+ARG GOFLAGS=-mod=mod
+ENV GOFLAGS=${GOFLAGS}
+
+RUN mkdir /src
+WORKDIR /src
+RUN CGO_ENABLED=0 go install -ldflags "-s -w -extldflags '-static'" github.com/go-delve/delve/cmd/dlv@latest
+COPY go.*  ./
+RUN go mod download
+COPY . ./
+
+FROM build as build-debug
+RUN GOARGS="-gcflags 'all=-N -l'" make probe
 
+FROM build as build-standard
+RUN make probe
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6 as debug
+COPY --from=build-debug /go/bin/dlv /src/probe/bin /stackrox/
+COPY --from=build-debug /src /src
+EXPOSE 7070
+ENTRYPOINT [ "/stackrox/dlv" , "--listen=:40000", "--headless=true", "--api-version=2", "--accept-multiclient", "exec", "/stackrox/probe"]
 CMD ["start"]
 
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6 as standard
```

```suggestion
FROM registry.access.redhat.com/ubi8/ubi-minimal:8.7 as standard
```

How about configuring dependabot to maintain it?

---

We can do this as a follow up, I don't have permissions to configure dependabot in the rpo.

It would also be good for the "main" image. When it's set up for the main image we can just add it for the probe image as well.

---

https://github.com/stackrox/acs-fleet-manager/pull/576